subtests/docker_cli/liverestore/liverestore.py

```diff
@@ -0,0 +1,185 @@
+r"""
+Summary
+---------
+
+Test live-restore: running container should survive docker daemon restart.
+
+Operational Summary
+----------------------
+
+#. Run a docker container
+#. Restart docker daemon
+#. Verify that container is listed under 'docker ps' and is still running.
+
+Operational Detail
+------------------
+
+Our test container outputs a time value (seconds since epoch) every
+second. We check that it's running by running `docker logs` repeatedly
+until we get a new line; timing out at five seconds.
+
+We *restart* the docker daemon, not separate stop and then start.
+The latter would give us more testing options, such as confirming
+continued output from the container, but there's also a lot more
+possibility of leaving our system in an undefined state if something
+goes wrong between stop and start.
+
+Prerequisites
+-------------
+
+This test is only applicable if docker daemon is run with --live-restore
+option; we abort with TestNAError if it isn't.
+
+live-restore also requires `KillMode=process` in the docker systemd
```

Comment:
It doesn't look like it. I've done all my testing with `MountFlags=slave` and all tests (including my manual ones) pass. I think it does require `fs.may_detach_mounts=1` but I haven't actually checked.